data-processing/pdf/process_pdf.py

```diff
@@ -0,0 +1,236 @@
+                bib_item_title = bib_item_titles.get(ref)
+                if bib_item_title:
+                    cited_paper = self.find_cited_paper(bib_item_title)
+                    if cited_paper:
+                        cited_paper_id = cited_paper['paperId']
+                        s2_ids_of_citations[ref] = cited_paper_id
+                        if cited_paper_id not in s2_data:
+                            authors = ','.join(a['name'] for a in cited_paper['authors'])
+                            s2_data[cited_paper_id] = SerializableReference(
+                                s2_id = cited_paper_id,
+                                arxivId=cited_paper.get('arxivId'),
+                                doi = cited_paper.get('doi'),
+                                title = cited_paper.get('title'),
+                                authors=authors,
+                                venue = cited_paper.get('venue'),
+                                year = cited_paper.get('year')
+                            )
+
+                citation_locations = set()
+                for box in self.get_bounding_boxes(spans):
+                        page = self.page_sizes[box.page]
+                        loc = CitationLocation(key=ref,
+                                     cluster_index=citation_index,
+                                     page=self.page_indices[box.page],
+                                     left=box.left / page['width'],
+                                     top=box.top / page['height'],
+                                     width=box.width / page['width'],
+                                     height=box.height / page['height'])
+                        citation_locations.add(loc)
+                locations.setdefault(ref, {})[citation_index] = citation_locations
+
+            citation_index += 1
+
+        return CitationData(
+            arxiv_id = None,
+            s2_id = pdf_hash,
+            citation_locations=locations,
+            key_s2_ids=s2_ids_of_citations,
+            s2_data = s2_data
+        )
+
+
+    def upload(self):
+        citations = self.get_citations()
+        upload_citations(citations, 'pdf-pipeline')
+        symbols = self.get_symbols()
+        upload_symbols(symbols,'pdf-pipeline')
+
+if __name__ == '__main__':
+    logging.basicConfig(level=logging.INFO)
+
+    pdf_hashes = ['8c53cd64e46cf382bf81ce2c4e0087ef31351919']
+    if sys.argv[1:]:
+        pdf_hashes = [s.strip() for s in open(sys.argv[1]).readlines()]
+
+    init_database_connections('public')
+    with TemporaryDirectory() as tempdir:
+        for pdf_hash in pdf_hashes:
+            start_time = time.time()
+            logging.info("Processing PDF %s", pdf_hash)
+            try:
+                pdf_file = "{}/{}.pdf".format(tempdir, pdf_hash)
+                try:
+                    subprocess.check_call([
+                        "aws", "s3", "cp", "s3://ai2-s2-pdfs/{}/{}.pdf".format(pdf_hash[:4], pdf_hash[4:]), pdf_file
```

Comment:
Does this mean this script should be run only within the AI2-internal AWS account?

Reply:
Well, if you run it using another account, the first S3 call will succeed, since that's a public bucket, but the fallback to `ai2-s2-pdfs-private` will fail.